ml/src/main/scala/ai/h2o/sparkling/ml/models/H2OTargetEncoderModel.scala

```diff
@@ -51,31 +54,36 @@ class H2OTargetEncoderModel(
   private def inputColumnNameToInternalOutputName(inputColumnName: String): String = inputColumnName + "_te"
 
   def transformTrainingDataset(dataset: Dataset[_]): DataFrame = {
-    val h2oContext = H2OContext.ensure("H2OContext needs to be created in order to use target encoding. Please create one as H2OContext.getOrCreate().")
+    val hc = H2OContext.ensure("H2OContext needs to be created in order to use target encoding. Please create one as H2OContext.getOrCreate().")
     val temporaryColumn = getClass.getSimpleName + "_temporary_id"
     val withIdDF = dataset.withColumn(temporaryColumn, monotonically_increasing_id)
     val flatDF = SchemaUtils.flattenDataFrame(withIdDF)
     val relevantColumns = getInputCols() ++ Array(getLabelCol(), getFoldCol(), temporaryColumn).flatMap(Option(_))
     val relevantColumnsDF = flatDF.select(relevantColumns.map(col(_)): _*)
-    val input = h2oContext.asH2OFrame(relevantColumnsDF)
+    val input = hc.asH2OFrameKeyString(relevantColumnsDF)
```

Comment:
Note: I think we should introduce asH2OFrame() which returns ai.h2o.sparkling.H2OFrame, but that is scope of different PR

Reply:
Yes, that would be great! Good idea.